web/packages/teleport/src/components/hooks/useInfiniteScroll/useInfiniteScroll.ts

```diff
@@ -58,14 +62,20 @@ export function useInfiniteScroll<T extends UnifiedResource>(
     recreateObserver();
   };
 
+  // Using layout effect instead of a regular one helps prevent sneaky race
+  // conditions. If we used a regular effect, the observer may be recreated
+  // after the current one (which, by now, may be tied to a stale state)
+  // triggers a fetch. Thus, the fetch would use stale state and may ultimately
+  // cause us to display incorrect data. This can be reproduced by rapidly
+  // changing filtering data on the resources list page.
```

Comment:
```suggestion
  // cause us to display incorrect data. (This issue can be reproduced by switching this to 
  // `useEffect` and rapidly changing filtering data on the resources list page).
```

Reply:
I notice that we're using `useLayoutEffect` here for what seems to be an identical reason to why we're using [this approach](https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes) in `useKeyBasedPagination`. Is there a simple way that we can modify one or the other to use a single approach for achieving this effect? Related question: is there particular reason to generally prefer one over the other?

Reply:
I don't know. Perhaps there is a better, simpler approach, but I think I reached a point of diminishing returns here with attempting to make this code as clear and robust as possible. I don't really remember why exactly, it was "just the way I did it".

Reply:
> Related question: is there particular reason to generally prefer one over the other?

I found this post to be useful. https://kentcdodds.com/blog/useeffect-vs-uselayouteffect

Reply:
@bl-nero If you don't know whether your design makes sense yet then feel free to convert this to a draft. Otherwise, come back and figure out the actual answer when you have more energy. I don't accept your apathetic response.